### DISCUSSION TITLE
RuleSet data type

### DISCUSSION THREAD

**Discussion by shaietz** [directus/directus#23909] [Feature Requests]
Hello! 👋 I think it could be really cool and useful to be able to have a set of rules / conditions (like the ones in the filters for example) as a data type. I have a particular usecase that i need this for but i think it could be useful for other cases as well.
The form ui for it can be exactly like the existing filtera and conditions ui.

I'd love to get some feedback on this and wether it is a good idea or not.

In case you do think this is a good idea i'd love to attempt to implement it myself 😅

**Comment by shaietz** [comment]
I now realize that this type already exists on things like user policies and conditional fields in forms so maybe it's as simple as just allowing the condition type to be a data type in the user defined item collections?

**Comment by rijkvanzanten** [comment]
I don't know if I wanna make that a true "datatype" in the system, as that's really a bit more low-level as how it's stored in the database. That being said, wouldn't you be able to just use a JSON field to store that information? 🤔

**Reply by shaietz** [reply]
I guess a better way to describe it is a rule set interface or filter interface.
Just to be clear i am talking about the   [filtering / conditional interface](https://docs.directus.io/user-guide/content-module/filters.html)

So the data itself would be json but with a custom rule set interface

This could also be an extension however i thought it made sense to be part of the main field types since there are alredy fields in the admin app with this type for instance the policy conditions or field conditions.
Maybe it could be an interface / display option for json fields? Or something simmilar to the repeater?

**Reply by rijkvanzanten** [reply]
Ah gotcha! Yeah it'd be "just" an interface that supports JSON fields. As a matter of fact, that's how the system-filter interface works currently! While not currently possible through the app, you could _technically_ already use that by setting the `interface` to `system-filter` in the API / database directly 👀

**Reply by shaietz** [reply]
Cool i didn't know that was an option, i'll try it!
However i do still think there is value in allowing users to create fields of this type because for example my use case is to allow a non texhnical user to define a set of rules that i would then use in my backend service in order to process data and so i would want to define by my self what fields and operators are available in the filter interface, which don't necessarily come from a directus collection (unless thats also possible 😅)
If (is far as i understand) the interface already exists and it's just a matter of enabling it in the admin app i think it could be usefull for other users who like me didn't know you could set it through the database

**Reply by rijkvanzanten** [reply]
Yes no yes no yes no. I don't disagree with you there, but at the same time this is the first time somebody has requested to use it outside of system config, so it does feel like a bit more of a niche thing. Don't want to overwhelm people with an interface that's most likely useless unless they have that specific use case 🤔 

That being said, I could see just allowing the system interfaces to be used as some collapsed sub-section so folks like yourself can just set it up when they want to recreate a system thing like that 🙂

**Reply by shaietz** [reply]
Ahh i see. I guess it really is a pretty niche use case if no ones requested it until now 😅
If you think other users could find use in a collapsed sub section of system interfaces than i'd be happy to work on it, however if it's just me than i might just make due with changing the types in the DB 🙂

**Reply by rijkvanzanten** [reply]
The trouble is that none of those system interfaces were built to be configurable, so none of them have interface options and might not work as expected in every use case 🤔 I don't know if just making them visible is going to help or hurt in most cases. There's also quite a few interfaces that really don't make sense outside of the Directus system use. F.e. the "enable mfa" interface won't be usable at all. My recommendation here would be to use the API or DB directly to set the interface to `system-filter` and go with that 🙂
